3landers.sol

```diff
@@ -0,0 +1,780 @@
+            emit PermanentURI(tokenURI(ids[i]), ids[i]);
+        }
+    }
+
+    function mintToken(uint256 amount, bytes calldata signature)
+        external
+        payable
+        nonReentrant
+        returns (bool)
+    {
+        require(!msg.sender.isContract(), "Contract is not allowed.");
+        require(
+            getState() == SaleState.PrivateSaleDuring ||
+                getState() == SaleState.PublicSaleDuring,
+            "Sale not available."
+        );
+
+        if (getState() == SaleState.PublicSaleDuring) {
+            require(
+                amount <= maxPublicSalePerTx,
+                "Mint exceed transaction limits."
+            );
+            require(
+                msg.value >= amount.mul(getPriceByMode()),
+                "Insufficient funds."
+            );
+            require(
+                totalSupply().add(amount).add(availableReserve()) <= maxSupply,
+                "Purchase exceed max supply."
+            );
+        }
+
+        if (getState() == SaleState.PrivateSaleDuring) {
+            require(isEIP712WhiteListed(signature), "Not whitelisted.");
+            require(
+                amount <= maxPrivateSalePerTx,
+                "Mint exceed transaction limits"
+            );
+            require(
+                _privateSaleClaimed[msg.sender] + amount <=
+                    maxWhitelistClaimPerWallet,
+                "Mint limit per wallet exceeded."
+            );
+            require(
+                totalPrivateSaleMinted.add(amount) <= privateSaleCapped,
+                "Purchase exceed private sale capped."
+            );
+
+            require(
+                msg.value >= amount.mul(getPriceByMode()),
+                "Insufficient funds."
+            );
+        }
+
+        if (
+            getState() == SaleState.PrivateSaleDuring ||
+            getState() == SaleState.PublicSaleDuring
+        ) {
+            _mintToken(msg.sender, amount);
+            if (getState() == SaleState.PublicSaleDuring) {
+                totalPublicMinted = totalPublicMinted + amount;
+            }
+            if (getState() == SaleState.PrivateSaleDuring) {
+                _privateSaleClaimed[msg.sender] =
+                    _privateSaleClaimed[msg.sender] +
+                    amount;
+                totalPrivateSaleMinted = totalPrivateSaleMinted + amount;
+            }
+            payable(_splitter).transfer(msg.value);
+        }
+
+        return true;
+    }
+
+    function setSeed(uint256 randomNumber) external onlyOwner {
+        randomseedRequested = true;
+        seed = randomNumber;
+        emit RandomseedFulfilmentManually(block.timestamp);
+    }
+
+    function setBaseURI(string memory baseURI) external onlyOwner {
+        _tokenBaseURI = baseURI;
+    }
+
+    function setDefaultURI(string memory defaultURI) external onlyOwner {
+        _defaultURI = defaultURI;
+    }
+
+    function requestChainlinkVRF() external onlyOwner {
```

Comment:
Chainlink does on chain random, it's used to set a seed value which essentially randomly distributes the metadata later on